.github/workflows/rocm-perf.yml

```
@@ -0,0 +1,128 @@
+
+    steps:
+      - name: Clean up old workdirs
+        run: |
+          docker run -v "$(pwd):/jax" ubuntu bash -c "chown -R $UID /jax/jax_rocm_perf_* || true"
+          rm -rf jax_rocm_perf_* || true
+          ls -l
+
+      - name: Print system info
+        run: |
+          whoami
+          printenv
+          df -h
+          rocm-smi || true
+
+      - name: Checkout JAX source
+        uses: actions/checkout@v4
+        with:
+          path: ${{ env.WORKSPACE_DIR }}
+
+      - name: Build JAX Docker Image
+        run: |
+          cd $WORKSPACE_DIR
+          python3 build/rocm/ci_build \
+            --rocm-version "$ROCM_VERSION" \
+            --base-docker "$BASE_IMAGE" \
+            --python-versions "$PYTHON_VERSION" \
+            --compiler=clang \
+            dist_docker \
+            --image-tag "$TEST_IMAGE"
+
+      - name: Checkout MaxText source
+        uses: actions/checkout@v4
+        with:
+          repository: ROCm/maxtext
+          ref: rv_jax
+          path: ${{ env.WORKSPACE_DIR }}/maxtext
+
+      - name: Launch container
+        run: |
+          docker run -d --name maxtext_container \
+            --network=host \
+            --device=/dev/kfd \
+            --device=/dev/dri \
+            --ipc=host \
+            --shm-size=64G \
+            --group-add=video \
+            --cap-add=SYS_PTRACE \
+            --security-opt seccomp=unconfined \
+            -v "$(pwd)/${{ env.WORKSPACE_DIR }}/maxtext:/maxtext" \
+            -w /maxtext \
+            "$TEST_IMAGE" \
+            tail -f /dev/null
+
+      - name: Run MaxText training and save logs
+        run: |
+          docker exec maxtext_container bash -c "pip install -r requirements.txt"
+          for config in \
+            MaxText/configs/models/gpu/llama2_7b_rocm.yml \
+            MaxText/configs/models/gpu/gemma_2b_rocm.yml \
+            MaxText/configs/models/gpu/gpt3_6b_rocm.yml \
+            MaxText/configs/models/gpu/mixtral_8x1b_rocm.yml; do
+            model_name=$(basename "$config" _rocm.yml)
+            echo "Running $model_name"
+            docker exec maxtext_container bash -c "python3 -m MaxText.train $config" \
+              | tee logs_${model_name}.log
+          done
+
+      - name: Analyze logs to compute median step time
+        run: |
+          pip install numpy
+          cat << EOF > analyze_logs.py
+import json, re, glob
+import numpy as np
+
+summary = {}
+for log in glob.glob("logs_*.log"):
+    model = log.replace("logs_", "").replace(".log", "")
+    times = []
+    with open(log) as f:
+        for line in f:
+            m = re.search(r"completed step: \d+, seconds: ([\d.]+)", line)
+            if m:
+                times.append(float(m.group(1)))
+    if times:
+        summary[model] = {
+            "median_step_time": round(float(np.median(times)), 3),
+            "steps_counted": len(times)
+        }
```

grab the parsed steps too with the summary

```suggestion
                times.append(float(m.group(1)))
    if times:
        step_info = list([{"step": n, "time": t} for n,t in enumerate(times)])
        summary[model] = {
            "steps": step_info,
            "median_step_time": round(float(np.median(times)), 3),
            "steps_counted": len(times)
        }
```

---

sounds good